core/services/directrequest/delegate.go

```diff
@@ -152,8 +154,9 @@ func (l *listener) Start() error {
 			},
 			NumConfirmations: l.minIncomingConfirmations,
 		})
-		l.shutdownWaitGroup.Add(2)
-		go l.run()
+		l.shutdownWaitGroup.Add(3)
+		go l.processOracleRequests()
```

Comment:
If a cancel request and a request come in around the same time, what happens if the cancel is processed first?

Reply:
Seems like the cancel could be missed. However, I think we don't even need the 2 goroutines - we can just have a single mailbox/goroutine since we already switch based on log type in the handleReceivedLogs

Reply:
based on an existing test, i thought cancel was meant to be processed concurrently. If we have just one mailbox+goroutine, then logs are processed fully serially and seems like there is no point in cancel, no?

Reply:
Ah I see, makes sense. In that case I think there might be a larger problem actually - the majority of the cancellation window would be while the request log is accumulating confirmations so I think we might need to subscribe with 0 confs and keep our own pool in the delegate so we can remove req logs if we see a cancel log?

Reply:
oh yeah interesting - indeed. I wonder if we could reuse the log pool from LB here, and have some reorg protection too.

Reply:
Is there any ticket/followup coming out of this?

Reply:
yup, 16779